src/extension.ts

```diff
@@ -11,29 +11,43 @@ export const activate = (context: vscode.ExtensionContext) => {
     return text;
   };
 
-  const disposable = vscode.commands.registerCommand("tempfile.newfile", () => {
-    try {
-      newfile();
-    } catch (e) {
-      if (e instanceof Error) {
-        vscode.window.showErrorMessage(e.message);
+  const register = (name: string) => {
+    const disposable = vscode.commands.registerCommand(name, () => {
+      try {
+        newfile();
+      } catch (e) {
+        if (e instanceof Error) {
+          vscode.window.showErrorMessage(e.message);
+        }
       }
-    }
-  });
-
-  vscode.workspace.openTextDocument();
+    });
+  
+    vscode.workspace.openTextDocument();
+  
+    context.subscriptions.push(disposable);
+  };
 
-  context.subscriptions.push(disposable);
+  ["default", "alt"].forEach(x => register(`tempfile.newfile.${x}`));
 };
 
-const newfile = () => {
+const newfile = async () => {
   const config = vscode.workspace.getConfiguration("tempfile");
 
   let pathTemplate = config.get<string>("newFilePath") ?? "";
   if (!pathTemplate.trim()) {
     pathTemplate = "{{tempdir}}/tempfile/{{YYYY}}{{MM}}{{DD}}_{{HH}}{{mm}}{{ss}}{{SSS}}.md";
   }
 
+  if (config.get<boolean>("ending") ?? false) {
+    pathTemplate = pathTemplate.replace(/(.*)\.[^.]+$/, "$1"); // remove possible set file endings
```

Comment:
Maybe this code doesn't work as expected for particular paths.

```js
"/aaa/bbb.ccc/ddd".replace(/(.*)\.[^.]+$/, "$1"); //'/aaa/bbb'
```

I think to process without regex is better.

```
// pseudo code
dname = dirname(path)
fname = basename(path)
if !fname.include(fname)
  return path

return join(dname, fname.without_exetension())
```

Reply:
Yeah i thought about that but what if we have for example `/some/dir/with.a.dot/filname-without-dot`

Reply:
Should be now fixed